src/swarm-scenario/Swarm/Game/Robot.hs

```diff
@@ -12,6 +12,13 @@
 -- A data type to represent robots.
 module Swarm.Game.Robot (
   -- * Robots data
+  RobotContextMember,
+  _robotID,
```

Comment:
Rather than exporting these explicit record fields, can we just use the lenses instead (which we are already exporting)?  e.g. instead of `_robotID r` we can write `r ^. robotID`.

Reply:
The `_robotID` and `_robotLocation` field exports are pretty much only needed for the `instantiateRobot` function in `Swarm.Game.Robot.Concrete`.  But it's unclear how lenses could substitute in that case.

Furthermore, yes, the `robotID` lens for the `_robotID` field is defined in `swarm-scenario`, but the `robotContext` and `machine` lenses are necessarily defined in `swarm-engine` because the `Store` and `CESK` types live there.